### PR TITLE
core,native: persist native contract via VM

### LINKED ISSUE #1060: Run OnPersist native calls as a proper script in a proper VM
Failing in `OnPersist` is dangerous:
```
2020-06-16T23:10:07.244+0300    INFO    blockchain persist completed    {"persistedBlocks": 27, "persistedKeys": 32, "headerHeight": 268150, "blockHeight": 69937, "took": "513.787µs"}
2020-06-16T23:10:07.467+0300    WARN    contract invocation failed      {"tx": "38046d6d1a58872f9295f322bb885118e8032ef3506fe9906d4dc1b61f1b3ddf", "block": 69940, "error": "error encountered at instruction 88 (ASSERT): ASSERT failed"}
panic: insufficient funds

goroutine 96 [running]:
github.com/nspcc-dev/neo-go/pkg/core/native.(*nep5TokenNative).addTokens(0xc0000bc1a0, 0xc0008eaf00, 0x1fb2d8f07c4df055, 0x8ceab25283c74179, 0xe5ee78, 0xc000804300)
        /home/rik/dev/neo-go/pkg/core/native/native_nep5.go:238 +0x379
github.com/nspcc-dev/neo-go/pkg/core/native.(*nep5TokenNative).burn(0xc0000bc1a0, 0xc0008eaf00, 0x1fb2d8f07c4df055, 0x8ceab25283c74179, 0xe5ee78, 0xc0008042e0)
        /home/rik/dev/neo-go/pkg/core/native/native_nep5.go:223 +0x102
github.com/nspcc-dev/neo-go/pkg/core/native.(*GAS).OnPersist(0xc0000bc1a0, 0xc0008eaf00, 0xfa7440, 0xc002c08fc0)
        /home/rik/dev/neo-go/pkg/core/native/native_gas.go:81 +0x108
github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).storeBlock(0xc00000a1e0, 0xc002de32c0, 0x0, 0x10)
        /home/rik/dev/neo-go/pkg/core/blockchain.go:638 +0x12ee
github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).AddBlock(0xc00000a1e0, 0xc002de32c0, 0x0, 0x0)
        /home/rik/dev/neo-go/pkg/core/blockchain.go:434 +0xee
github.com/nspcc-dev/neo-go/pkg/network.(*blockQueue).run(0xc0001e8de0)
        /home/rik/dev/neo-go/pkg/network/blockqueue.go:48 +0x158
created by github.com/nspcc-dev/neo-go/pkg/network.(*Server).Start
        /home/rik/dev/neo-go/pkg/network/server.go:179 +0x2b5
```
We're not supposed to fail there, but at the same time we have another problems with missing proper VM context like #924 and notifications/application log not being generated for the block as it should be.

So in the end we should create the same `OnPersist` script as C# node does and run it in a VM and treat the result of this script the same way as a regular transaction.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #1060 .

After native contracts are deployed, single persist script is created
and executed at the start of every block persisting.

### CODE REVIEW

File: pkg/core/blockchain.go [hunk 1]
```diff
@@ -576,6 +576,8 @@ func (bc *Blockchain) storeBlock(block *block.Block) error {
 		v.SetPriceGetter(getPrice)
 		if bc.config.FreeGasLimit > 0 {
 			v.GasLimit = bc.config.FreeGasLimit + tx.SystemFee
+		} else {
+			v.GasLimit = tx.SystemFee
```
Comment (roman-khimov) [inline pkg/core/blockchain.go:580]: I think we may not touch it until #1064.

File: pkg/core/blockchain.go [hunk 2]
```diff
@@ -556,6 +556,15 @@ func (bc *Blockchain) storeBlock(block *block.Block) error {
 		return err
 	}
 
+	if block.Index > 0 {
+		systemInterop := bc.newInteropContext(trigger.System, cache, block, nil)
+		v := SpawnVM(systemInterop)
+		v.LoadScriptWithFlags(bc.contracts.GetPersistScript(), smartcontract.AllowModifyStates|smartcontract.AllowCall)
+		if err := v.Run(); err != nil {
+			bc.log.Warn("can't persist native contracts", zap.Error(err))
```
Comment (roman-khimov) [inline pkg/core/blockchain.go:564]: I think we should fail persisting the block then. And there also should be a wrapper cached DAO for these cases. In general, it's not that different from transaction execution handling.

Comment (fyrchik) [reply pkg/core/blockchain.go:564]: Fixed.

Comment (fyrchik) [reply pkg/core/blockchain.go:564]: What do you mean by `wrapper cached DAO`? We use already created cache and `interop.NewContext` takes care about wrapping.

Comment (roman-khimov) [reply pkg/core/blockchain.go:564]: OK.

Comment (roman-khimov) [reply pkg/core/blockchain.go:564]: But I don't see it being persisted though.

**Comment by fyrchik** [comment]
Added `AppExecResult` saving.
Dropped a comment about extending `getapplicationlog` in #1062 .
I am not sure it is needed in event stream, as I see no usecase and it will be always the same (besides consumed gas):
`HALT`, 2 `true` items on stack (# of native contracts), `trigger.System`.

**Comment by roman-khimov** [comment]
But there will be notifications on burned/minted gas and we need them.

File: pkg/core/blockchain.go (hunk 2 above)
Comment (fyrchik) [reply pkg/core/blockchain.go]: Oh, yeah, fixed.

**Comment by roman-khimov** [comment]
So what about `bc.processNEP5Transfer`?

**Comment by fyrchik** [comment]
@roman-khimov I am working on it. RN `block.ConsensusData.PrimaryIndex` is not set properly in tests which makes it hard to make changes in RPC tests.

**Pull request merged**